routes/compat/wakatime/v1/heartbeat.go

```diff
@@ -54,15 +62,23 @@ func (h *HeartbeatHandler) Get(w http.ResponseWriter, r *http.Request) {
 		w.Write([]byte("bad date"))
 		return
 	}
+
 	timezone := user.TZ()
-	rangeFrom, rangeTo := utils.StartOfDay(date).In(timezone), utils.EndOfDay(date).In(timezone)
+	rangeFrom, rangeTo := utils.StartOfDay(date.In(timezone)), utils.EndOfDay(date.In(timezone))
 
-	heartbeats, err := h.heartbeatSrvc.GetAllWithin(rangeFrom, rangeTo, user)
+	heartbeats, err := h.heartbeatSrvc.GetAllWithinUTC(rangeFrom, rangeTo, user)
 	if err != nil {
 		w.WriteHeader(http.StatusInternalServerError)
 		w.Write([]byte(conf.ErrInternalServerError))
 		conf.Log().Request(r).Error("failed to retrieve heartbeats - %v", err)
 		return
 	}
-	utils.RespondJSON(w, r, http.StatusOK, heartbeats)
+
+	res := HeartbeatsResult{
+		Data:     heartbeats,
```

Comment:
One issue regarding data compatibility again. Of course, the endpoints won't be 100 % compatible. But one thing, that people might find really annoying when coding against Wakapi and WakaTime API at the same time is the different format of `time`. WakaTime returns a timestamp like `1643562171.033` (1), while this endpoint returns `2022-01-21T00:38:46+01:00` (2), for instance. `models.CustomTime` is, admittedly, a bit weird in that is unmarshals from (1), but marshals to (2). One solution would be to simply make `time` a `float` and then map to the time's timestamp for every heartbeat before returning it. Or you might also come up with a better way.

Maybe you can also make use of [`v1.HeartbeatEntry`](https://github.com/muety/wakapi/blob/master/models/compat/wakatime/v1/heartbeat.go) somehow, so we don't have two separate compat heartbeat view models?

Reply:
Please apologize all these change requests.

Reply:
> then map to the time's timestamp for every heartbeat before returning it

Can you please elaborate on this?

I've had a look at a sample output from the wakatime api. 
Besides the `time` field, I think a large issue is the various guids, which is non-trivial to replicate (requiring some large migrations, I assume?). However, if we don't use guids, then information will be lost if `v1.HeartbeatEntry` is used. 
So I'd be inclined to change `model.CustomTime` `date` to `float`, if you have no concerns with that?

Reply:
I've experimented more with this, a couple of possible solutions:
1. Duplicate/change `v1.HeartbeatEntry`, to add additional fields so that information isn't lost. 
2. Duplicate `models.Heartbeat`, change type of `Time` to another `CustomTimeFloat` which deserialize to float and add conversion function to map to this new type which will only be used for this endpoint. 
3. Change `Time` of `models.Heartbeat` to another `CustomTimeFloat`. Downside: Had a taste of this in converting to float, is more reference changes. 
4. Redefine CustomTime to be struct `{ Time: time.Time; UnmarshallToFloat64 bool }`, loop over heartbeats to set that boolean, change Unmarshal function. Downside: I've tried this. It's even more reference changes. 

Now that I understand this issue a bit more, seems like (1) and (2) are what you suggested, will try.

Reply:
What about using `v1.HeartbeatEntry` but changing `Time` field to `models.CustomTimeFloat` (or so), which is the same as `models.CustomTime`, but besides unmarshalling _from_ float also marshals _to_ float? The missing fields (e.g. `editor`, etc.) are, strictly speaking, not in WakaTime's view model, so shouldn't be included in ours either. But if you still want to include them I wouldn't mind of you just extended the `v1.HeartbeatEntry` by those few fields.

Reply:
New example response:
```json
{
  "data": [
    {
      "id": "1",
      "branch": "",
      "category": "",
      "entity": "/home/user1/dev/project1/main.go",
      "is_write": true,
      "language": "Go",
      "project": "wakapi",
      "time": 1640995199,
      "type": "file",
      "user_id": "testuser",
      "machine_name_id": "",
      "user_agent_id": "PostmanRuntime/7.29.0"
    },
    {
      "id": "2",
      "branch": "",
      "category": "",
      "entity": "/home/user1/dev/project1/main.go",
      "is_write": true,
      "language": "Go",
      "project": "wakapi",
      "time": 1641077999,
      "type": "file",
      "user_id": "testuser",
      "machine_name_id": "",
      "user_agent_id": "PostmanRuntime/7.29.0"
    }
  ],
  "end": "2022-01-01T23:00:00Z",
  "start": "2021-12-31T23:00:00Z",
  "timezone": "Europe/Berlin"
}
```